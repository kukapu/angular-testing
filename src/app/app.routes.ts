import { Routes } from '@angular/router';
import { CounterComponent } from './basic/counter/counter.component';

export const routes: Routes = [
  {
    path: 'basic/counter',
    component: CounterComponent
  },
  {
    path: '**',
    redirectTo: 'basic/counter'
  }
];
