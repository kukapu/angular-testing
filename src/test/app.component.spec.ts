import { TestBed } from '@angular/core/testing';
import { AppComponent } from '../app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
    }).compileComponents();
  });

  test('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  test(`should have the 'angular-testing' title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component.title).toEqual('angular-testing');
  });

  // test('should render title', () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   fixture.detectChanges();
  //   const compiled = fixture.nativeElement as HTMLElement;
  //   const component = fixture.componentInstance;

  //   const h1 = compiled.querySelector('h1');
  //   expect(compiled.querySelector('h1')?.textContent).toContain(component.title);
  // });

  // test('should match snapshot', () => {
  //   const fixture = TestBed.createComponent(AppComponent);
  //   fixture.detectChanges();
  //   const compiled = fixture.nativeElement as HTMLElement;

  //   expect(compiled).toMatchSnapshot();
  // })
});
