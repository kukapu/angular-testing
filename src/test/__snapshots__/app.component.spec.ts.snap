// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<div
  id="root3"
>
  <h1>
    angular-testing
  </h1>
  <hr />
  <hr />
</div>
`;
